pkg/virt-api/webhooks/mutating-webhook/mutators/vm-mutator_test.go

```diff
@@ -350,6 +367,829 @@ var _ = Describe("VirtualMachine Mutator", func() {
+					},
+				},
+				TypeMeta: k8smetav1.TypeMeta{
+					Kind: "PersistentVolumeClaim",
+				},
+			}
+			_, err := virtClient.CoreV1().PersistentVolumeClaims(vm.Namespace).Create(context.Background(), pvc, k8smetav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			dvWithSourcePVC = &v1beta1.DataVolume{
+				ObjectMeta: k8smetav1.ObjectMeta{
+					Name:      "dvWithSourcePVC",
+					Namespace: vm.Namespace,
+				},
+				TypeMeta: k8smetav1.TypeMeta{
+					Kind: "DataVolume",
+				},
+				Spec: v1beta1.DataVolumeSpec{
+					Source: &v1beta1.DataVolumeSource{
+						PVC: &v1beta1.DataVolumeSourcePVC{
+							Name:      pvc.Name,
+							Namespace: pvc.Namespace,
+						},
+					},
+				},
+			}
+			_, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(vm.Namespace).Create(context.Background(), dvWithSourcePVC, k8smetav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			dsWithSourcePVC = &v1beta1.DataSource{
+				ObjectMeta: k8smetav1.ObjectMeta{
+					Name:      dsWithSourcePVCName,
+					Namespace: vm.Namespace,
+				},
+				TypeMeta: k8smetav1.TypeMeta{
+					Kind: "DataSource",
+				},
+				Spec: v1beta1.DataSourceSpec{
+					Source: v1beta1.DataSourceSource{
+						PVC: &v1beta1.DataVolumeSourcePVC{
+							Name:      pvc.Name,
+							Namespace: pvc.Namespace,
+						},
+					},
+				},
+			}
+			_, err = virtClient.CdiClient().CdiV1beta1().DataSources(vm.Namespace).Create(context.Background(), dsWithSourcePVC, k8smetav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			dsWithAnnotations = &v1beta1.DataSource{
+				ObjectMeta: k8smetav1.ObjectMeta{
+					Name:      dsWithAnnotionsName,
+					Namespace: vm.Namespace,
+					Annotations: map[string]string{
+						instancetype.DefaultInstancetypeAnnotation:     defaultInferedName,
+						instancetype.DefaultInstancetypeKindAnnotation: defaultInferedKind,
+						instancetype.DefaultPreferenceAnnotation:       defaultInferedName,
+						instancetype.DefaultPreferenceKindAnnotation:   defaultInferedKind,
+					},
+				},
+				TypeMeta: k8smetav1.TypeMeta{
+					Kind: "DataSource",
+				},
+				Spec: v1beta1.DataSourceSpec{
+					Source: v1beta1.DataSourceSource{
+						PVC: &v1beta1.DataVolumeSourcePVC{
+							Name:      pvc.Name,
+							Namespace: pvc.Namespace,
+						},
+					},
+				},
+			}
+			_, err = virtClient.CdiClient().CdiV1beta1().DataSources(vm.Namespace).Create(context.Background(), dsWithAnnotations, k8smetav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+		})
+
+		DescribeTable("should infer defaults from VolumeSource and PersistentVolumeClaim", func(instancetypeMatcher, expectedInstancetypeMatcher *v1.InstancetypeMatcher, preferenceMatcher, expectedPreferenceMatcher *v1.PreferenceMatcher) {
+			vm.Spec.Instancetype = instancetypeMatcher
+			vm.Spec.Preference = preferenceMatcher
+			vm.Spec.Template.Spec.Volumes = append(vm.Spec.Template.Spec.Volumes, v1.Volume{
```

Comment:
Is there anything to append to, or can you directly assign the slice of volumes?